src/privileged/autoplay/api.js

```
@@ -0,0 +1,337 @@
+
+  submitTelemetryPing(data) {
+    console.log("submitTelemetryPing to server");
+    console.log(data);
+    const telOptions = { addClientId: true, addEnvironment: true };
+    return TelemetryController.submitExternalPing("block-autoplay", data, telOptions);
+  }
+
+  async constructPayload(payloadType) {
+    const payload = {
+      id: this.pingId++,
+      clientId: await getTelemetryId(),
+      branch: this.branch,
+      type: payloadType,
+    };
+    switch (payloadType) {
+      case "counts":
+        payload.counters = {
+          totalPages: this.domainUserVisited.size,
+          totalPagesAM: this.domainWithAutoplay.size,
+          totalBlockedAudibleMedia: this.getBlockedAudibleMediaCount(),
+        };
+        break;
+      case "prompt":
+        payload.promptResponse = [];
+        while (this.promptResponses.length > 0) {
+          const data = this.promptResponses.shift();
+          payload.promptResponse.push(data);
+        }
+        break;
+      case "settings":
+        payload.settingsChanged = [];
+        while (this.settingChanges.length > 0) {
+          const data = this.settingChanges.shift();
+          payload.settingsChanged.push(data);
+        }
+        break;
+      default:
+        console.log("Error : incorrect payload type");
+        break;
+    }
+    return payload;
+  }
+
+  getBlockedAudibleMediaCount() {
+    const option = Ci.nsITelemetry.DATASET_RELEASE_CHANNEL_OPTIN;
+    const scalar = Services.telemetry.snapshotScalars(option, false);
+    // the count we get is accumulated with session life time, so we need to calculate
+    // the count within the period between sending the ping.
+    const accumulatedCount = scalar.content["media.autoplay_would_not_be_allowed_count"];
+    if (!Number.isInteger(accumulatedCount)) {
+      return 0;
+    }
+
+    const count = accumulatedCount - this.blockedAudibleAutoplayCount;
+    if (count < 0) {
+      console.log("### Error : count should not be negative.");
```

What are you trying to do here? Just logging "Error" seems pretty inconsequential, either this is an issue (in which case you should throw an error or return 0) or it's not.

---

Thank you, I would throw an error here.